src/linagora.esn.contact/app/sidebar/sidebar.controller.js

```diff
@@ -150,7 +150,23 @@ function ContactSidebarController(
     var categories = contactAddressbookDisplayService.categorizeDisplayShells(self.displayShells);
 
     self.userAddressbooks = categories.userAddressbooks;
-    self.sharedAddressbooks = categories.sharedAddressbooks;
+    self.sharedAddressbooks = _getUniqueSharedAddressbooks(categories.sharedAddressbooks);
     self.virtualAddressbooks = categories.virtualAddressbooks;
   }
+
+  function _getUniqueSharedAddressbooks(addressbooks) {
+    if (!addressbooks) return;
+
+    const uniqueAddressbookList = addressbooks.reduce((acc, current) => {
+      const { source: { href } } = current.shell;
```

Comment:
Is this equivalent to `const href = current.shell.href;`?
I find this syntax hard to read.

Reply:
it's just an Object destructuring assignment

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring